Rhys/main.cpp

```diff
@@ -0,0 +1,195 @@
+                    return "Access Permitted";
+                }
+            }
+        }
+    }
+    return "Access Denied";
+}
+
+vector<string> retrievePosts(string username) {
+    if (!dataLoaded) {
+        return {"Please load data first"};
+    }
+    vector<string> visiblePosts;
+    for (const Post& post : posts) {
+        if (post.userId != username || 
+            (post.visibility == "friend" && find_if(users.begin(), users.end(), [&post, &username](const User& u) {
+                return u.username == post.userId && find(u.friends.begin(), u.friends.end(), username) != u.friends.end();
+            }) != users.end())) {
+            visiblePosts.push_back(post.postId);
+        }
+    }
+    return visiblePosts;
+}
+
+vector<string> searchUsersByLocation(string state){
+    if (!dataLoaded) {
+        return {"Please load data first"};
+    }
+    vector<string> matchedUsers;
+    for(const User& user : users){
+        if(user.state == state){
+            matchedUsers.push_back(user.displayName);
+        }
+    }
+    return matchedUsers;
+}
+
+int main(){
+    string userFile, postFile;
+    int choice = 0;
+    bool loop = true;
+    cout << "Please Select a Numbered Option From the Following:\n";
+    cout << "1. Load input data\n";
+    cout << "2. Check visibility\n";
+    cout << "3. Retrieve posts\n";
+    cout << "4. Search users by location\n";
+    cout << "5. Exit\n";
+    while(loop){
+        cin >> choice;
+        cout << endl;
+        switch (choice) {
+            case 1:
+                cout << "Enter user file path: ";
+                cin >> userFile;
+                cout << "Enter post file path: ";
+                cin >> postFile;
+                loadUserData(userFile);
+                loadPostData(postFile);
+                cout << "Data from user and post files successfully loaded!\n";
+                break;
+            case 2:
+                if (!dataLoaded) {
+                    cout << "Please load data first" << endl;
+                } else {
+                    string postId, username;
+                    cout << "Enter a postID: ";
+                    cin >> postId;
+                    cout << "Enter the username: ";
+                    cin >> username;
+                    cout << checkVisibility(postId, username) << endl;
+                }
+                break;
+            case 3:
+                if (!dataLoaded) {
+                    cout << "Please load data first" << endl;
+                } else {
+                    string username;
+                    cout << "Enter a username: ";
+                    cin >> username;
+                    vector<string> posts = retrievePosts(username);
+                    for (const string& postId : posts) {
+                        cout << postId << endl;
+                    }
+                }
+                break;
+            case 4:

```

Comment:
*Option 4 works flawlessly*
+When a valid location is provided the users are provided accordingly. 
+When a invalid location is provided there is no response.
^Suggestion, add a invalid location error